dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_fields.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React from 'react';
+import { EuiButtonIcon, EuiPanel, EuiSpacer, EuiText, EuiTitle } from '@elastic/eui';
+import { ConfigListEntry } from '../../../../../../../../common/types/explorer';
+import { VIS_CHART_TYPES } from '../../../../../../../../common/constants/shared';
+import {
+  AGGREGATIONS,
+  CUSTOM_LABEL,
+  GROUPBY,
+} from '../../../../../../../../common/constants/explorer';
+
+export const DataConfigPanelFields = ({
+  list,
+  sectionName,
+  visType,
+  addButtonText,
+  handleServiceAdd,
+  handleServiceRemove,
+  handleServiceEdit,
+}: any) => {
+  const isHeatMapAddButton = (name: string) => {
+    if (visType === VIS_CHART_TYPES.HeatMap) {
+      if (name === AGGREGATIONS) return list?.length === 1;
+      return !(list?.length < 2);
+    }
+    return false;
+  };
+
+  return (
+    <div className="panel_section">
+      <EuiTitle size="xxs" className="panel_title">
+        <h3>{sectionName}</h3>
+      </EuiTitle>
+      <EuiSpacer size="s" />
+      {list !== undefined &&
```

Comment:
Have we checked if the list is not an array? If we have many places that requires list to be existing and an array, we could lift empty checking before any calculation based on list data.

Reply:
Done